Shared.js

```diff
@@ -1,156 +1,242 @@
+            if (typeof (wdc[FUNCTION_NAMES[ii]]) !== 'function') {
+                throw new Error(`The connector did not define the required function: ${FUNCTION_NAMES[ii]}`);
+            }
+        }
+
+        console.log('Connector registered');
+
+        this.globalObj._wdc = wdc;
+        this._wdc = wdc;
+    }
 
-Shared.prototype._initTriggerFunctions = function() {
-  this.privateApiObj.triggerInitialization = this._triggerInitialization.bind(this);
-  this.privateApiObj.triggerSchemaGathering = this._triggerSchemaGathering.bind(this);
-  this.privateApiObj.triggerDataGathering = this._triggerDataGathering.bind(this);
-  this.privateApiObj.triggerShutdown = this._triggerShutdown.bind(this);
-}
+    /**
+     * @returns {Undefined}
+     */
+    _initTriggerFunctions () {
+        this.privateApiObj.triggerInitialization = this._triggerInitialization.bind(this);
+        this.privateApiObj.triggerSchemaGathering = this._triggerSchemaGathering.bind(this);
+        this.privateApiObj.triggerDataGathering = this._triggerDataGathering.bind(this);
+        this.privateApiObj.triggerShutdown = this._triggerShutdown.bind(this);
+    }
 
-// Starts the WDC
-Shared.prototype._triggerInitialization = function() {
-  this._wdc.init(this.privateApiObj._initCallback);
-}
+    /**
+     * Starts the WDC
+     * @returns {Undefined}
+     */
+    _triggerInitialization () {
+        this._wdc.init(this.privateApiObj._initCallback);
+    }
 
-// Starts the schema gathering process
-Shared.prototype._triggerSchemaGathering = function() {
-  this._wdc.getSchema(this.privateApiObj._schemaCallback);
-}
+    /**
+     * Starts the schema gathering process
+     * @returns {Undefined}
+     */
+    _triggerSchemaGathering () {
+        this._wdc.getSchema(this.privateApiObj._schemaCallback);
+    }
 
-// Starts the data gathering process
-Shared.prototype._triggerDataGathering = function(tablesAndIncrementValues) {
-  if (tablesAndIncrementValues.length != 1) {
-    throw ("Unexpected number of tables specified. Expected 1, actual " + tablesAndIncrementValues.length.toString());
-  }
-
-  var tableAndIncremntValue = tablesAndIncrementValues[0];
-  var isJoinFiltered = !!tableAndIncremntValue.filterColumnId;
-  var table = new Table(
-    tableAndIncremntValue.tableInfo, 
-    tableAndIncremntValue.incrementValue, 
-    isJoinFiltered, 
-    tableAndIncremntValue.filterColumnId || '', 
-    tableAndIncremntValue.filterValues || [],
-    this.privateApiObj._tableDataCallback);
-
-  this._wdc.getData(table, this.privateApiObj._dataDoneCallback);
-}
+    /**
+     * Starts the data gathering process
+     * @param {Array} tablesAndIncrementValues
+     * @returns {undefined}
+     */
+    _triggerDataGathering (tablesAndIncrementValues) {
+
+        if (tablesAndIncrementValues.length !== 1) {
+            throw new Error(`Unexpected number of tables specified. Expected 1, actual  ${tablesAndIncrementValues.length.toString()}`);
```

Comment:
In `${tablesAndIncrementValues.length.toString()}`, the `.toString()` is not necessary anymore

Reply:
done